src/OAuth2/Storage/Cassandra.php

```diff
@@ -423,43 +423,54 @@ public function getEncryptionAlgorithm($client_id = null)
         if (!is_array($userDetails)) {
             return false;
         }
 
-        $claims = explode(' ', trim($claims));
         $userClaims = array();
-
-        // for each requested claim, if the user has the claim, set it in the response
-        $validClaims = explode(' ', self::VALID_CLAIMS);
-        foreach ($validClaims as $validClaim) {
-            if (in_array($validClaim, $claims)) {
-                if ($validClaim == 'address') {
-                    // address is an object with subfields
-                    $userClaims['address'] = $this->getUserClaim($validClaim, $userDetails['address'] ?: $userDetails);
-                } else {
-                    $userClaims = array_merge($userClaims, $this->getUserClaim($validClaim, $userDetails));
-                }
-            }
+        $scopeValues = array_intersect(explode(' ', self::VALID_CLAIMS), explode(' ', $scope));
+        foreach ($scopeValues as $scopeValue) {
+            $userClaims = array_merge($userClaims, $this->getUserClaim($scopeValue, $userDetails));
         }
 
         return $userClaims;
     }
 
-    protected function getUserClaim($claim, $userDetails)
+    protected function getUserClaim($scopeValue, $userDetails)
     {
         $userClaims = array();
-        $claimValuesString = constant(sprintf('self::%s_CLAIM_VALUES', strtoupper($claim)));
+        $claimValuesString = constant(sprintf('self::%s_CLAIM_VALUES', strtoupper($scopeValue)));
         $claimValues = explode(' ', $claimValuesString);
 
-        foreach ($claimValues as $value) {
-            if ($value == 'email_verified') {
-                $userClaims[$value] = $userDetails[$value]=='true' ? true : false;
-            } else {
-                $userClaims[$value] = isset($userDetails[$value]) ? $userDetails[$value] : null;
+        if ($scopeValue === self::SCOPE_ADDRESS) {
+            if (isset($userDetails[self::SCOPE_ADDRESS]) && is_array($userDetails[self::SCOPE_ADDRESS])) {
+                $userDetails = $userDetails[self::SCOPE_ADDRESS];
+            }
+
+            $addressClaims = array();
+
+            foreach ($claimValues as $claimValue) {
+                if (isset($userDetails[$claimValue])) {
+                    $addressClaims[$claimValue] = $userDetails[$claimValue];
+                }
+            }
+
+            if (count($addressClaims)) {
+                $userClaims[self::SCOPE_ADDRESS] = $addressClaims;
+            }
+        }
+        else {
+            foreach ($claimValues as $claimValue) {
+                if (isset($userDetails[$claimValue])) {
+                    if (in_array($claimValue, array(self::CLAIM_EMAIL_VERIFIED, self::CLAIM_PHONE_NUMBER_VERIFIED), true)) {
+                        $userDetails[$claimValue] = (is_string($userDetails[$claimValue]) && !strcasecmp($userDetails[$claimValue], 'true')) || ((is_numeric($userDetails[$claimValue]) || is_bool($userDetails[$claimValue])) && (bool)$userDetails[$claimValue]);
```

Comment:
Please break this out across multiple lines and add comments, so the logic is clear.